[BOOM-6247] 1 click upgrade v 4.0.0

> This issue has been migrated from the Forge. Read the [original ticket here](http://forge.prestashop.com/browse/BOOM-6247).

- _**Reporter:**_ PN
- _**Created at:**_ Tue, 14 Aug 2018 09:47:56 +0200


<p>Hello,</p>

<p>I try update PS 1.7.3.2 to 1.7.4.2</p>

<p> </p>

<p><b><em>Here you are some issues after finished:</em></b></p>

<p> </p>

<p>Go to admin page – only white page is present.</p>

<p>Solution: go to the modules ps_facetedsearch folder and delete vendor folder - now is BO page OK</p>

<p> </p>

<p>First problem: after done upgrade all main menu link in BO are with message: you cannot access…</p>

<p>Solution: table ps_access is empty – I loaded table ps_access from backups – now I have access but</p>

<p>Here is second problem: go to modules, click configure - you do not have the permission for configure…</p>

<p>Solution: ???</p>

<p>Next is all translations in BO main menu missing – remain only English.</p>

<p>I have question why update DB (sql commands) start from <b>1.7.0.0 to 1.7.4.2</b>? I starting update PS version <b>1.7.3.2</b>.</p>

<p>Thank you</p>

<p>Best regards</p>

<p>PN</p>

<p><b>Errors:</b></p>

<p><span class="error">&#91;INTERNAL&#93;</span> home/www/domain.cz/www/update.domain.cz/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/Loader/YamlFileLoader.php line 667 - Call to undefined method Symfony\Component\Yaml\Parser::parseFile():</p>

<p><span class="error">&#91;INTERNAL&#93;</span> /home/www/domain.cz/www/domain.cz/admin/autoupgrade/latest/install/upgrade/php/ps1700_stores.php line 35 - Invalid argument supplied for foreach()</p>

<p><span class="error">&#91;ERROR&#93;</span> PHP 1.7.0.0 /* PHP:ps1700_stores(); */ </p>
        
* How to reproduce the issue ?

Click upgrade

When the update module finishes I get the message [ERROR] PHP 1.7.0.0 / * PHP: ps1700_stores (); * /, I can access the back and the front but after deactivating some modules for compatibility I suddenly get this error:
Fatal error: Uncaught -> Smarty: Plugin tag 'escape' already registered <- thrown in /test/vendor/smarty/smarty/libs/sysplugins/smarty_internal_method_registerplugin.php on line 43

And I do not know where exactly it comes from.

--- a/install-dev/upgrade/sql/1.7.0.0.sql
+++ b/install-dev/upgrade/sql/1.7.0.0.sql
@@ -117,6 +117,14 @@ CREATE TABLE `PREFIX_authorization_role` (
   UNIQUE KEY (`slug`)
 ) ENGINE=ENGINE_TYPE DEFAULT CHARSET=utf8;
 
+/* Add missing access entries on certain installation */
+INSERT INTO `PREFIX_access` (`id_profile`, `id_tab`, `view`, `add`, `edit`, `delete`)
+  SELECT 1, `id_tab`, 0, 0 ,0 ,0
+  FROM `PREFIX_tab` WHERE `id_tab` NOT IN (SELECT `id_tab` FROM `PREFIX_access` WHERE `id_profile` = 1);
+INSERT INTO `PREFIX_module_access` (`id_profile`, `id_module`, `view`, `configure`, `uninstall`)
+  SELECT 1, `id_module`, 1, 0, 0
+  FROM `PREFIX_module` WHERE `id_module` NOT IN (SELECT `id_module` FROM `PREFIX_module_access` WHERE `id_profile` = 1);
+
 /* Create a copy without indexes to make ID updates without conflict. */
 CREATE TABLE `PREFIX_access_old` AS SELECT * FROM `PREFIX_access`;
 DROP TABLE `PREFIX_access`;

